ldap/servers/slapd/log.c

```diff
@@ -3013,137 +3205,991 @@ log__open_accesslogfile(int logfile_state, int locked)
+        PR_Lock(lbi->lock);
+        log_flush_buffer(lbi, SLAPD_SECURITY_LOG, 1 /* sync to disk now */);
+        PR_Unlock(lbi->lock);
+    }
+}
+
+/******************************************************************************
+* write in the security log
+******************************************************************************/
+static int
+vslapd_log_security(const char *log_data)
+{
+    char buffer[SLAPI_LOG_BUFSIZ];
+    time_t tnl;
+    int32_t blen = TBUFSIZE;
+    int32_t rc = LDAP_SUCCESS;
+
+#ifdef SYSTEMTAP
+    STAP_PROBE(ns-slapd, vslapd_log_security__entry);
+#endif
+
+    /* We do this sooner, because that we can use the message in other calls */
+    if ((blen = PR_snprintf(buffer, SLAPI_LOG_BUFSIZ, "%s\n", log_data)) == -1) {
+        log__error_emergency("vslapd_log_security, Unable to format message", 1, 0);
+        return -1;
+    }
+
+#ifdef SYSTEMTAP
+    STAP_PROBE(ns-slapd, vslapd_log_security__prepared);
+#endif
+    tnl = slapi_current_utc_time();
+    log_append_security_buffer(tnl, loginfo.log_security_buffer, buffer, blen);
+
+#ifdef SYSTEMTAP
+    STAP_PROBE(ns-slapd, vslapd_log_security__buffer);
+#endif
+
+    return (rc);
+}
+
+int
+slapi_log_security(Slapi_PBlock *pb, const char *event_type, const char *msg)
+{
+    Connection *pb_conn = NULL;
+    Slapi_Operation *operation = NULL;
+    Slapi_DN *bind_sdn = NULL;
+    ber_tag_t method = LBER_DEFAULT;
+    char method_and_mech[TBUFSIZE] = {0};
+    char authz_target[TBUFSIZE] = {0};
+    char utc_time[TBUFSIZE] = {0};
+    char binddn[BUFSIZ] = {0};
+    char *mod_dn = NULL;
+    char *saslmech = NULL;
+    const char *target_dn = NULL;
+    char *server_ip = NULL;
+    char *client_ip = NULL;
+    char local_time[TBUFSIZE] = {0};
+    char *msg_text = (char *)msg;
+    int32_t ltlen = TBUFSIZE;
+    int external_bind = 0;
+    int ldap_version = 3;
+    int isroot = 0;
+    int rc = 0;
+    uint64_t conn_id = 0;
+    int32_t op_id = 0;
+    json_object *log_json = NULL;
+
+    if (!(loginfo.log_security_state & LOGGING_ENABLED) ||
+        !loginfo.log_security_fdes ||
+        !loginfo.log_security_file)
+    {
+        return 0;
+    }
+
+    slapi_pblock_get(pb, SLAPI_CONNECTION, &pb_conn);
+    slapi_pblock_get(pb, SLAPI_OPERATION, &operation);
+    slapi_pblock_get(pb, SLAPI_BIND_TARGET_SDN, &bind_sdn);  // this is not correct for mod ops
+    slapi_pblock_get(pb, SLAPI_ORIGINAL_TARGET_DN, &target_dn);
+    slapi_pblock_get(pb, SLAPI_REQUESTOR_ISROOT, &isroot);
+    slapi_pblock_get(pb, SLAPI_BIND_METHOD, &method);
+    slapi_pblock_get(pb, SLAPI_BIND_SASLMECHANISM, &saslmech);
+
+    conn_id = pb_conn->c_connid,
+    op_id = operation->o_opid,
+    client_ip = pb_conn->c_ipaddr;
+    server_ip = pb_conn->c_serveripaddr;
+    ldap_version = pb_conn->c_ldapversion;
+    if (saslmech) {
+        external_bind = !strcasecmp(saslmech, LDAP_SASL_EXTERNAL);
+    }
+
+    slapi_log_err(SLAPI_LOG_ALERT, "MARK", "event: %p\n", event_type); // DEBUG
```

Comment:
This logs 
```
[20/Jul/2022:15:32:22.010146800 +0000] - ALERT - MARK - event: 0x563919d14386
```
Should be removed, I guess?

Reply:
Whoops yeah that needs to come out.